test/type_check/macros_test.exs

```diff
@@ -26,5 +26,32 @@ defmodule TypeCheck.MacrosTest do
         assert Enum.all?(x, fn elem -> is_integer(elem) end)
       end
     end
+
+    test "support unquote fragments" do
+      defmodule UnquoteFragmentSpec do
+        defmacro my_macro(name) do
+          quote bind_quoted: [name: name], location: :keep do
+
+            use TypeCheck
```

Comment:
At least in my examples, `use TypeCheck` is not needed.

Reply:
`use TypeCheck` needs to be included into the final module one way or another, because of two reasons:
1. to ensure `spec!`, `type!` etc. can be found (although manually calling `import TypeCheck.Macros` does the same)
2. to setup an `@before_compile`-hook in which all the magic (wrapping the functions with versions that perform the checks) happens. 

Whether this is done in the module your macro is used in or inside the quote inside your macro does not really matter, although usually the former should probably be preferred because it makes it more clear that TypeCheck is used, and allows you to pass the module-specific options to TypeCheck.

I'll alter the test to move the statement to the actual example module because I think it is clearer. Thanks for the comment! :+1:

Reply:
Ah, I missed that it was not in the actual implementation. Thank you for the additional information and suggested usage.